### PR TITLE
Retrieving zones added and all hosts from all projects.

### PR DESCRIPTION
**Pull request by mateusz-szczyrzyca**
##### ISSUE TYPE

 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: contrib/inventory/gce_googleapiclient.py [hunk 1]
```diff
@@ -137,10 +158,16 @@ def main(args):
     zone = args['--zone']
     api_version = args['--api-version']
 
-    instances = get_instances(project_id=project,
-                              zone=zone,
-                              api_version=api_version)
+    projects = get_all_projects()
 
+    instances = []
+    for project in projects:
```
Comment (bart2) [inline contrib/inventory/gce_googleapiclient.py]: we can do:
```
for project in get_all_projects():
```

**Review by bart2 — COMMENTED** [review]
small code enhancement

File: contrib/inventory/gce_googleapiclient.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
     {script_name} [options]
 
 Options:
+    --account ACCOUNT_NAME --account=ACCOUNT    Account billing name
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: use either `ACCOUNT_NAME` or `ACCOUNT` on both. also 'Account billing name' -> 'Billing account name'

File: contrib/inventory/gce_googleapiclient.py [hunk 3]
```diff
@@ -43,28 +44,49 @@
 """.format(script_name=basename(argv[0]), envvar_prefix=ENV_PREFIX)
 
 
-def get_all_projects(api_version='v1'):
+
+def get_all_billing_projects(account_billing_name, api_version='v1'):
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: `account_billing_name` doesn't make sense. use `billing_account_name` instead.

File: contrib/inventory/gce_googleapiclient.py [hunk 4]
```diff
@@ -43,28 +44,49 @@
...
 
     credentials = GoogleCredentials.get_application_default()
 
-    service = discovery.build('cloudresourcemanager', api_version,
+    service = discovery.build('cloudbilling', 
+                              version=api_version,
                               credentials=credentials)
     # pylint: disable=no-member
-    request = service.projects().list()
+    request = service.billingAccounts().projects().list(name=account_billing_name)
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: same as before.

File: contrib/inventory/gce_googleapiclient.py [hunk 5]
```diff
@@ -43,28 +44,49 @@
...
-        for project in response['projects']:
-            project_ids.append(project['projectId'])
-
         # pylint: disable=no-member
-        request = service.projects().list_next(previous_request=request,
-                                               previous_response=response)
+        request = service.billingAccounts().projects().list_next(previous_request=request,
+                                                                 previous_response=response)
+
+        for projectBillingInfo in response['projectBillingInfo']:
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: variable name should not have UPPER CASE characters, replace `projectBillingInfo` with `project_billing_info`.
i know it was like that on the sample i've shared with you, but it was already fixed on mine.

File: contrib/inventory/gce_googleapiclient.py [hunk 6]
```diff
@@ -134,12 +159,17 @@ def get_inventory(instances):
 
 def main(args):
     project = args['--project']
-    zone = args['--zone']
+    zone = args['--zone']   
     api_version = args['--api-version']
+    account_billing_name = args['--account']
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: same with `account_billing_name`.

File: contrib/inventory/get_all_available_instances.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
+package main
```
Comment (maurofaccenda) [inline contrib/inventory/get_all_available_instances.go]: why this .go file?

**Review by maurofaccenda — CHANGES_REQUESTED** [review]

File: contrib/inventory/gce_googleapiclient.py [hunk 8]
```diff
@@ -33,38 +33,60 @@
     {script_name} [options]
 
 Options:
-    -a API_VERSION --api-version=API_VERSION    The API version used to connect to GCE [default: v1]
-    -c CONFIG_FILE --config=CONFIG_FILE         Path to the config file (see docoptcfg docs) [default: ./gce_googleapiclient.ini]
-    -l --list                                   List all hosts (needed by Ansible, but actually doesn't do anything)
-    -p PROJECT --project=PROJECT                The GCE project where you want to get the inventory
-    -z ZONE --zone=ZONE                         The GCE zone where you ant to get the inventory
+    --account ACCOUNT_NAME --account=ACCOUNT_NAME  Account billing name
```
Comment (maurofaccenda) [inline contrib/inventory/gce_googleapiclient.py]: Account billing name -> Billing account name

**Review by maurofaccenda — APPROVED** [review]

**Review by maurofaccenda — APPROVED** [review]

**Pull request merged**